File: frontend/src/types/problem.ts
export interface Problem {
    id: number;
    title: string;
    slug: string;
    description: string;
    difficulty: 'EASY' | 'MEDIUM' | 'HARD';
    timeLimitMs: number;
    memoryLimitMb: number;
    isActive: boolean;
}

export interface TestCaseRequest {
    input: string;
    expectedOutput: string;
    isHidden: boolean;
    weightage: number;
}

export interface ProblemRequest {
    title: string;
    slug: string;
    description: string;
    difficulty: string;
    timeLimitMs: number;
    memoryLimitMb: number;
    testCases: TestCaseRequest[];
}

File: backend/src/main/java/com/codejudge/backend/model/Problem.java
package com.codejudge.backend.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "problems")
@Data
@NoArgsConstructor
public class Problem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(unique = true, nullable = false)
    private String slug;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private String difficulty; // EASY, MEDIUM, HARD

    @Column(name = "time_limit_ms")
    private Integer timeLimitMs = 1000;

    @Column(name = "memory_limit_mb")
    private Integer memoryLimitMb = 256;

    @Column(name = "is_active")
    private Boolean isActive = true;

    @ManyToOne
    @JoinColumn(name = "created_by")
    private User createdBy;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}
